pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1254,6 +1254,32 @@ static __always_inline bool is_compat(struct task_struct *task)
 #endif
 }
 
+static __always_inline int get_syscall_id_from_regs(struct pt_regs *regs)
+{
+#if defined(bpf_target_x86)
+    int id = READ_KERN(regs->orig_ax);
+#elif defined(bpf_target_arm64)
+    int id = READ_KERN(regs->syscallno);
+#endif
+    return id;
+}
+
+static __always_inline struct pt_regs *get_task_pt_regs(struct task_struct *task)
```

Comment:
Thats a very nice approach. I think it needs better documentation (I see this comes from task_pt_regs() from in kernel). To document why you're rounding up current kernel stack (I believe you're going after thread_info->task_struct and picking up pt_regs from task_struct, right ?).

We currently don't support other page sizes (other than 4k) but, if and when we do, we will need to calculate THREAD_SIZE dynamically.

Reply:
@yanivagman thats actually a good thing, should we worry about different page sizes ? There are many servers with hugepages around there (specially those trying to avoid TLB lookup bottlenecks for data passing etc). Of course the majority is 4k page sized (tends to change in arm64 world at least).

Reply:
Actually, this code (x86 part) isn't new. We had this code up until tracee v0.6.3, and removed it after adding syscall data stashing.
We need to return it now that we want to do some performance optimizations, like @NDStrahilevitz mentioned.



> I see this comes from task_pt_regs() from in kernel

That's correct

> We currently don't support other page sizes (other than 4k) but, if and when we do, we will need to calculate THREAD_SIZE dynamically.

Agree. We should document this here

Reply:
> @yanivagman thats actually a good thing, should we worry about different page sizes ? There are many servers with hugepages around there (specially those trying to avoid TLB lookup bottlenecks for data passing etc). Of course the majority is 4k page sized (tends to change in arm64 world at least).

We can use https://pkg.go.dev/os#Getpagesize in userspace to initialize a corresponding value in the config map, then use it here. WDYT?

Reply:
Hum, that is a nice approach... hopefully verifiers will accept that. Im okay in having the 4k default for 0.8.2 (for performance improvements) and have an issue to get the page size for this (and possibly other sensitive points).